BEPs/BEP-346.md

```diff
@@ -0,0 +1,121 @@
+
+### 4.2 Implementation
+
+#### 4.2.1. Change To Current Model
+- [ObjectInfo](https://docs.bnbchain.org/greenfield-docs/docs/guide/storage-provider/modules/common/proto/#objectinfo-proto):
+  ObjectInfo is used for object metadata stored in Greenfield. Below fields are added.
+  - `updating`: Indicate whether an object is being updated. It means the user has initiated an update request, and the SP has not committed the change to Greenfield yet.
+  - `updated_at`: The timestamp that the user initiated to update the object. The storage price is determined by it, and a new store fee needs to be calculated.
+  - `version`: The current version of the object. It can be used to locate the correct version of object data in SP.
+
+#### 4.2.1. New Model And Message
+- `ShadowObjectInfo`: It is introduced to store user updated object metadata, the life cycle for a shadowObjectInfo is 
+  started when users trigger the update request on Greenfield, once SP has the updated object data stored and confirm 
+  on Greenfield, it will be removed.
+  - `operator`: The account who initiates update object request, can be the owner or any account that has been granted permission to update object content.
+  - `id`: The id of the object, it is the same as the id in ObjectInfo.
+  - `payload_size`: The updated object size.
+  - `checksums`: The updated checksums of pieces stored in storage providers.
+  - `updated_at`: The timestamp that user initiated to update object request.
+  - `version`: The expected next version of the object.
+- `MsgUpdateObjectContent`: A new message for users' update object content transaction. The new object's payload_size,
+  checksums along with approval from primary SP's approval need to be specifcied.
+```protobuf
+message MsgUpdateObjectContent {
+  option (cosmos.msg.v1.signer) = "operator";
+  // operator defines the account address of object uploader
+  string operator = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+  // bucket_name defines the name of the bucket where the object is stored.
+  string bucket_name = 2;
+  // object_name defines the name of object
+  string object_name = 3;
+  // payload_size defines size of the object's payload
+  uint64 payload_size = 4;
+  // content_type defines a standard MIME type describing the format of the object.
+  string content_type = 5;
+  // primary_sp_approval defines the approval info of the primary SP which indicates that primary sp confirm the user's request.
+  common.Approval primary_sp_approval = 6;
+  // expect_checksums defines a list of hashes which was generate by redundancy algorithm.
+  repeated bytes expect_checksums = 7;
+}
+```
+
+## 4.3 Workflow
+
+The steps to update an object are illustrated below. To users, it is similar when compared to the current create-upload object process.
+
+![](./assets/BEP-346/workflow.png)
+
+1. Users obtain approval from the primary SP where the object is stored and send the transaction to Greenfield with the 
```

Comment:
Why is this step necessary? Can it be eliminated?

Reply:
not required any more